deep_qa/data/instances/tuple_inference_instance.py

```diff
@@ -0,0 +1,443 @@
+    def pad(self, max_lengths: Dict[str, int]):
+        """
+        Pads (or truncates) each indexed tuple in answers_indexed and background_indexed to specified lengths,
+        using the superclass methods ``pad_sequence_to_length`` (for padding number of options, number of tuples,
+        and number of slots per tuple) and ''pad_word_sequence`` (for padding the words in each slot), with the
+        following exception:
+            When padding the number of answer options, if there are more answers provided, here we don't
+            remove options, we raise an exception because we don't want to risk removing the correct answer in
+            a multiple choice setting.
+        """
+        desired_num_options = max_lengths['num_options']
+        desired_num_question_tuples = max_lengths['num_question_tuples']
+        desired_num_background_tuples = max_lengths['num_background_tuples']
+        desired_num_slots = max_lengths['num_slots']
+        desired_question_slot_length = max_lengths['question_slot_length']
+        desired_background_slot_length = max_lengths['background_slot_length']
+        desired_word_character_length = None
+        if 'max_word_character_length' in max_lengths:
+            desired_word_character_length = max_lengths['max_word_character_length']
+        # Pad the number of answers. Note that while we use the superclass method when we need to add empty
+        # answer_options, we don't want to remove answer options.
+        num_options = len(self.answers_indexed)
+        if num_options > desired_num_options:
+            raise Exception("Case of too many answer options ({0} provided) isn't currently "
+                            "handled.".format(num_options))
+        elif num_options < desired_num_options:
+            # Add empty answer option(s).
+            self.answers_indexed = self.pad_sequence_to_length(self.answers_indexed,
+                                                               desired_num_options,
+                                                               lambda: [],
+                                                               truncate_from_right=False)
+        # Pad the number of tuples per option.
+        for answer_index in range(len(self.answers_indexed)):
+            self.answers_indexed[answer_index] = self.pad_sequence_to_length(self.answers_indexed[answer_index],
+                                                                             desired_num_question_tuples,
+                                                                             lambda: [],
+                                                                             truncate_from_right=False)
+        # Pad each of the tuples for slot length and words per slot
+        for answer_index in range(len(self.answers_indexed)):
+            self.answers_indexed[answer_index] = [self.pad_tuple(answer_tuple,
+                                                                 desired_num_slots,
```

Comment:
Do you know what the average values are for the question slot length vs. the background slot length?  If they are not very different, I think it would make the code a lot simpler to just use the same length for both.